### PR TITLE
User model

### PR DESCRIPTION
**Pull request by scottPlusPlus**
Adding a rough POC for a UserStore, currently only works in-memory. 

Also needed to bang on the configs a bit to pull in the shared Data-Model folder to the typescript build process

### CODE REVIEW

File: backend/src/auth/test/TestMockUserStore.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { UserModel } from "../data_model/UserModel";
```
Comment (ArendPeter) [inline backend/src/auth/test/TestMockUserStore.ts:1]: Thanks for adding tests!

File: backend/src/auth/data_model/UserModel.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { Uid } from "../../../../domain_model/Uid"
+
+
+export class UserModel {
+    id: Uid;
+    email: Email;
+    hashedPassword: string;
+    salt: string;
+
+    constructor(data: Pick<UserModel, "id" | "email" | "hashedPassword" | "salt">) {
```
Comment (ArendPeter) [inline backend/src/auth/data_model/UserModel.ts:11]: I haven't used Pick before, I'm assuming that his requires that id, email, hashedPassword, and salt are given as separate parameters and then it's combined into the data parameter?

File: backend/src/auth/data_model/UserModel.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        this.id = data.id;
+        this.email = data.email;
+        this.hashedPassword = data.hashedPassword;
```
Comment (ArendPeter) [inline backend/src/auth/data_model/UserModel.ts:14]: At what point will we generate the hashedPassword from the raw password? Could we take the raw password as a parameter and do the hashing here?

**Comment by ArendPeter** [comment]
I think the c52dcf92e8e64063350e700aa9ec5ad8b692f845 commit made it into both reviews

**Review by ArendPeter — APPROVED** [review]
I mostly had some questions for my understanding, but overall it looks good to me, nice work!

File: backend/src/auth/test/TestMockUserStore.ts (hunk 1 above)
Comment (scottPlusPlus) [reply backend/src/auth/test/TestMockUserStore.ts:1]: One thing we'll need to do is find a good testing framework

File: backend/src/auth/data_model/UserModel.ts (hunk 2 above)
Comment (scottPlusPlus) [reply backend/src/auth/data_model/UserModel.ts:11]: My understanding is its a subset of fields from the initial type (in this case UserModel).  This was effectively a shortcut from defining an interface with the same fields as the UserModel, and the constructor will take anything (including an anonymous object) that has all the fields in the Pick

File: backend/src/auth/data_model/UserModel.ts (hunk 3 above)
Comment (scottPlusPlus) [reply backend/src/auth/data_model/UserModel.ts:14]: We'd do that when a user first registers, or changes their password.  I didn't stick that in the constructor as I imagine those being explicit methods.

**Pull request merged**